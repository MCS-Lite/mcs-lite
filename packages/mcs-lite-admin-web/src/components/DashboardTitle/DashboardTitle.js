import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Heading from "mcs-lite-ui/lib/Heading";

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  padding-bottom: 5px;
  margin-bottom: 10px;
  border-bottom: 1px solid ${props => props.theme.color.grayDark};
`;

const DashboardTitle = ({ title, children }) => (
  <Wrapper>
    <Heading level={4}>{title}</Heading>
    {children}
  </Wrapper>
);

DashboardTitle.displayName = "DashboardTitle";
DashboardTitle.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node
};
export default DashboardTitle;
