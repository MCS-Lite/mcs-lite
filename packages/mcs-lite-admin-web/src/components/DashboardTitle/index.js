import DashboardTitle from './DashboardTitle';

export default DashboardTitle;
