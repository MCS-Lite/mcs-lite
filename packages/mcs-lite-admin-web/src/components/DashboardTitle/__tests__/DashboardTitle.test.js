import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DashboardTitle from '../';

it('should renders <DashboardTitle> correctly without children', () => {
  const wrapper = shallow(<DashboardTitle title="title" />);

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('should renders <DashboardTitle> correctly with children', () => {
  const wrapper = shallow(
    <DashboardTitle title="title">
      children
    </DashboardTitle>,
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
