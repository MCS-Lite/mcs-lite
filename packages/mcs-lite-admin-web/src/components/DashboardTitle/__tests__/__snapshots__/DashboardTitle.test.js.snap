// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders <DashboardTitle> correctly with children 1`] = `
<styled.div>
  <Heading
    level={4}
  >
    title
  </Heading>
  children
</styled.div>
`;

exports[`should renders <DashboardTitle> correctly without children 1`] = `
<styled.div>
  <Heading
    level={4}
  >
    title
  </Heading>
</styled.div>
`;
